### DISCUSSION TITLE
NobleAnimation from multiple gifs

### DISCUSSION THREAD

**Discussion by willmartian** [NobleRobot/NobleEngine#41] [Ideas and Feature Requests]
Hi, nice library! Really helping me get up and running with Playdate dev.

I am working on a game, and I have separate .gifs for different animations: `idle.gif`, `sleeping.gif`, `running.gif`, ... etc.

Is there a way for NobleAnimation/NobleSprite to be constructed from multiple file paths? I want each of these .gifs to correspond to an animation state. If this is not currently possible, would you be open to a PR that added it? Or do you suggest reformatting the assets?

**Comment by Mark-LaCroix** [comment]
That's an interesting use-case, I'd love to support it. I think the way I'd want it to look is that the user could include a gif for the default state, and would add additional gifs when calling `addState()` on a `NobleSprite.animation`, rather than including a bunch of gifs in the constructor when creating the `NobleSprite` itself.

If you can put together a PR for it, I say go for it!

There is another PR in review (just meaning I haven't gotten around to pulling it yet) right now #38 that makes some useful changes to `NobleSprite` that might align with this. If you do make a PR, take a look at how it would fit with that one, but I will merge and address any conflicts that come up so don't worry about that.
